extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -275,6 +296,14 @@ export class AmpAd3PImpl extends AMP.BaseElement {
       user.assert(!this.isInFixedContainer_,
           '<amp-ad> is not allowed to be placed in elements with ' +
           'position:fixed: %s', this.element);
+      /** detect ad containers, add the list to element as a new attribute */
+      if (!this.element.getAttribute('data-amp-container-element')) {
+        const containerList_ = getAdContainerList(this.element);
+        if (containerList_) {
+          this.element.setAttribute('data-amp-container-element',
```

Comment:
I see. https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/0.1/amp-ad-3p-impl.js#L283
`#getIframe` function from `3p-frame.js` will get all attributes from `amp-ad` and pass them through url.